lib/matplotlib/colors.py

```diff
@@ -506,6 +506,9 @@ def to_rgba_array(c, alpha=None):
 
     if alpha is not None:
         rgba[:, 3] = alpha
+    if isinstance(c, Sequence):
+        rgba[:, 3] = [0 if isinstance(cc, str) and cc == 'none'
+                      else rgba[i, 3] for i, cc in enumerate(c)]
```

Comment:
AFAICT, we can move this in: `to_rgba("none")` returns the correct alpha. Only the explicit alpha can mess this  up. Thus, we don't have to spend the extra effort for cases without explicit alpha.

Also using a numpy mask should be somewhat faster and clearer.

```suggestion
        if isinstance(c, Sequence):
            # ensure that an explicit alpha does not overwrite full transparency for "none"
            none_mask = np.array([isinstance(cc, str) and cc == "none" for cc in c])
            rgba[:, 3][none_mask] = 0
```